pkg/wallet/crypto_box.go

```diff
@@ -33,8 +33,13 @@ type CryptoBox struct {
 }
 
 // NewCryptoBox creates a CryptoBox which provides crypto box encryption using the given wallet's keypairs
-func NewCryptoBox(w *BaseWallet) *CryptoBox {
-	return &CryptoBox{w: w}
+func NewCryptoBox(w Crypto) *CryptoBox {
+	wa, ok := w.(*BaseWallet)
+	if !ok {
+		return &CryptoBox{}
+	}
```

Comment:
This only happens if you try to encrypt using a mock wallet as key provider - how should the CryptoBox behave when it's expecting to have a wallet that it can get private keys from? This brings us back around to all of the problems with my previous attempt at crypto refactoring - for the mock wallet to mock BaseWallet providing private keys to CryptoBox, we'd need to do some ugly refactoring, which doesn't mesh with the plans for 0.1.1.

Reply:
that's fine then.. we'll refactor this later